tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc

```diff
@@ -31,6 +31,55 @@ namespace mlir {
 namespace xla_lhlo {
 namespace {
 
+struct DotOpConverter : public OpRewritePattern<DotOp> {
+  using OpRewritePattern<DotOp>::OpRewritePattern;
+  // Supports only rank-2 tensors.
+  LogicalResult matchAndRewrite(DotOp op,
+                                PatternRewriter& rewriter) const override {
+    const auto& lhs = op.lhs();
+    const auto& rhs = op.rhs();
+    const auto& lhs_type = lhs.getType().template cast<MemRefType>();
+    const auto& rhs_type = rhs.getType().template cast<MemRefType>();
+    const auto& element_type = lhs_type.getElementType();
+    const auto& shape_lhs = lhs_type.getShape();
+    const auto& shape_rhs = rhs_type.getShape();
```

Comment:
Don't use `auto` when the type isn't obvious.
Also these are all Value types and should be used as reference.
Finally avoid `const` for IR objects (https://mlir.llvm.org/docs/Rationale/UsageOfConst/)

Reply:
@joker-eph Can you confirm MLIR style also applies here (since we are in TF)? Surrounding code does use const references here.

Reply:
@joker-eph Changes have been made.